### ISSUE TITLE
Failed to replace libeay32.lib  ssleay32.lib  with latest version compiled by myself

### ISSUE THREAD

**Issue by connLAN** [SoftEtherVPN/SoftEtherVPN#1155]
### Prerequisites

* [ yes] Can you reproduce?
* [ yes ] Are you running the latest version of SoftEtherVPN?

**SoftEther version:**
**Component:** [Server, Client, Bridge, etc.]
**Operating system:** [Windows]
**Architecture:** [64 bit]

### Description

[ replace libeay32.lib  ssleay32.lib with latest version ]

[ Failed to compile, ]
Mayaqua.lib error LNK2019  unresolved external symbol - function OPENSSL_sk_value()  ... 

A lot of such errors about openssl library functions. 

### Steps to reproduce

1. Clone SoftEtherVPN, cd src, run BuildAll.cmd , and all is ok;
2, Replace libeay32.lib and ssleay32.lib with I just compiled;
3, LNK2019 error occured.

I just compiled out  libeay32.lib and ssleay32.lib of version HEAD 、 V1.0.2、V1.0.1  with MS2017 Community edition，none of them can be compiled into SoftEtherVPN.  Is there any important thing I has ignored?  Apperciate for any help.

**Comment by davidebeatrici** [comment]
Could you paste the error(s) here, please?

**Comment by connLAN** [comment]
The project "SeLow" is not selected for building in solution configuration "Release |Win32".
. \\Proxy.c : warning C4819: This file contains code that cannot be found on the current code page (936). ). Please save the file in Unicode format to prevent data loss!
. \Proxy.c(10): warning C4013: "WinConnectEx3 "Undefined; assume external return int
. \Proxy.c(10): warning C4047: "return": " SOCK *" is not the same indirect level as "int".
libeay32.lib(ebcdic.obj) : warning LNK4221: Public symbol not found; archive member will be inaccessible
libeay32.lib(ecp_nistp224.obj) : warning LNK4221: Public symbol not found; archive members will not be accessible
libeay32.lib(ecp_nistp256.obj) : warning LNK4221: Public symbol not found; archive members will not be accessible
libeay32.lib(ecp_nistp521.obj) : warning LNK4221: Public symbol not found; archive members will not be accessible
libeay32.lib(ecp_nistputil.obj) : warning LNK4221: Public symbol not found; archive members will not be accessible
libeay32.lib(rand_egd.obj) : warning LNK4221: Public symbol not found; archive members will not be accessible

 "D:\src_github\softethervpn\src\SEVPN.sln" (Rebuild target) (1) ->
(Mayaqua:Rebuild target) -> 
  . \\Proxy.c : warning C4819: This file contains code that cannot be found on the current code page (936). ). Please save the file in Unicode format to prevent data loss!
  . \Proxy.c(10): warning C4013: "WinConnectEx3 "Undefined; assume external return int
  . \Proxy.c(10): warning C4047: "return": " SOCK *" is not the same indirect level as "int".
  libeay32.lib(ebcdic.obj) : warning LNK4221: Public symbol not found; archive member will be inaccessible
  libeay32.lib(ecp_nistp224.obj) : warning LNK4221: Public symbol not found; archive members will not be accessible
  libeay32.lib(ecp_nistp256.obj) : warning LNK4221: Public symbol not found; archive members will not be accessible
  libeay32.lib(ecp_nistp521.obj) : warning LNK4221: Public symbol not found; archive members will not be accessible
  libeay32.lib(ecp_nistputil.obj) : warning LNK4221: Public symbol not found; archive members will not be accessible
  libeay32.lib(rand_egd.obj) : warning LNK4221: Public symbol not found; archive members will not be accessible
 

LIBCMT.lib(sprintf.obj) : error LNK2005. sprintf has been defined in Mayaqua.lib(dso_win32.obj): sprintf has been defined in Mayaqua.lib(dso_win32.obj)
LIBCMT.lib(vsprintf.obj) : error LNK2005: _ vsprintf_l already in Mayaqua.lib(dso_win32.obj) define
LIBCMT.lib(printf.obj) : error LNK2005. printf has been defined in Mayaqua.lib(pqueue.obj): printf has been defined in Mayaqua.lib(pqueue.obj)
Mayaqua.lib(Network.obj) : error LNK2019: Unresolvable external symbol OPENSSL_sk_num, which is in the function sk_SSL_ Referenced in CIPHER_num
Mayaqua.lib(Encrypt.obj) : error LNK2001: Unable to parse external symbol OPENSSL_sk_num
Mayaqua.lib(Network.obj) : error LNK2019: Unresolvable external symbol OPENSSL_sk_value, which is in the function sk_SSL_ Referenced in CIPHER_value
Mayaqua.lib(Encrypt.obj) : error LNK2001: Unable to parse external symbol OPENSSL_sk_value
Mayaqua.lib(Network.obj) : error LNK2019: Unresolvable external symbol OPENSSL_sk_free, which is in the function sk_SSL_ Referenced in CIPHER_free
Mayaqua.lib(Network.obj) : error LNK2019: Unresolvable external symbol X509_STORE_CTX_get0_cert, which is in the function Referenced in SslCertVerifyCallback
Mayaqua.lib(Network.obj) : error LNK2019: Unresolvable external symbol SSL_CTX_set_options, which is in the function Cited in NewSSLCtx
Mayaqua.lib(Network.obj) : error LNK2019: Unresolvable external symbol TLS_method, which is referenced in the function NewSSLCtx
Mayaqua.lib(Network.obj) : error LNK2019: Unresolvable external symbol SSL_get1_supported_ciphers, which is in the function Referenced in GetCipherList
Mayaqua.lib(Network.obj) : error LNK2019: Unresolvable external symbol TLS_server_method, which is in the function Referenced in GetCipherList
Mayaqua.lib(Network.obj) : error LNK2019: Unresolvable external symbol TLS_client_method, which is in the function Referenced in StartSSLEx
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol OPENSSL_sk_new_null, which is in the function sk_ Referenced in ASN1_OBJECT_new_null
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol OPENSSL_sk_push, which is in the function sk_ASN1_ Referenced in OBJECT_push
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol OPENSSL_sk_pop_free, which is in the function sk_ Referenced in ASN1_OBJECT_pop_free
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol HMAC_CTX_free, which is in the function Enc_tls1_P_ is referenced in hash
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol HMAC_CTX_new which is in the function Enc_tls1_P_ is referenced in hash
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol EVP_MD_CTX_free, which is used in the function FreeMd by the quote
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol OPENSSL_thread_stop, which is in the function Referenced in FreeOpenSSLThreadState
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol DH_get0_key, which is referenced in the function DhNew
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol DH_set0_pqg, which is referenced in the function DhNew
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol EVP_MD_CTX_new, which in the function NewMdEx was quote
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol RSA_get0_key, which is in the function Referenced in RsaPublicToBuf
Mayaqua.lib(Secure.obj) : error LNK2001: cannot External symbol for parsing RSA_get0_key
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol EVP_PKEY_get0_RSA, which is in the function Referenced in RsaPublicToBuf
Mayaqua.lib(Secure.obj) : error LNK2001: cannot External symbol for parsing EVP_PKEY_get0_RSA
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol X509_get0_notAfter, which is in the function NewX509 cited in
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol X509_get0_notBefore, which is in the function Cited in NewX509
Mayaqua.lib(Encrypt.obj) : error LNK2019: Unresolvable external symbol ASN1_STRING_get0_data, which is in the function Referenced in X509ToX
Mayaqua.lib(Secure.obj) : error LNK2019: unable to The external notation of the parser, RSA_get0_crt_params, is used in the function WriteSecKey is referenced in
Mayaqua.lib(Secure.obj) : error LNK2019: unable to RSA_get0_factors is the external symbol of the parsing, which is used in the WriteSecKey cited in
Mayaqua.lib(ui_openssl.obj) : error LNK2001 : Unresolvable external symbol __acrt_iob_func
Mayaqua.lib(pqueue.obj) : error LNK2019: unable to The external symbol of the parsing __acrt_iob_func, which is referenced in the function printf
Mayaqua.lib(cryptlib.obj) : error LNK2001: Unresolvable external symbol __acrt_iob_func
Mayaqua.lib(t1_enc.obj) : error LNK2001: Unable to External symbols for parsing __acrt_iob_func
Mayaqua.lib(txt_db.obj) : error LNK2001: Unable to External symbols for parsing __acrt_iob_func
Mayaqua.lib(pqueue.obj) : error LNK2019: unable to The external symbol of the parsing __stdio_common_vfprintf, which is used in the function _ Referenced in vfprintf_l
Mayaqua.lib(t1_enc.obj) : error LNK2001: Unable to Parsed external symbols __stdio_common_vfprintf
Mayaqua.lib(txt_db.obj) : error LNK2001: cannot Parsed external symbols __stdio_common_vfprintf
Mayaqua.lib(ui_openssl.obj) : error LNK2001 : Unresolved external symbol __stdio_common_vfprintf
Mayaqua.lib(dso_win32.obj) : error LNK2019: Unresolvable external symbol __stdio_common_vsprintf, which is in the function _ Referenced in vsnprintf_l
Mayaqua.lib(cryptlib.obj) : error LNK2001: Unable to parse external symbols __stdio_common_vsprintf
Mayaqua.lib(ecp_nistz256.obj) : error LNK2001: Unresolved External Symbols __report_ rangecheckfailure
Mayaqua.lib(b_dump.obj) : error LNK2001: cannot External notation for parsing __report_rangecheckfailure
Mayaqua.lib(obj_dat.obj) : error LNK2019: Unresolvable external symbol __report_rangecheckfailure, which is used in the function Referenced in OBJ_create_objects
Mayaqua.lib(b_print.obj) : error LNK2001: Unresolvable external symbol __report_rangecheckfailure
Mayaqua.lib(t1_lib.obj) : error LNK2001: cannot External notation for parsing __report_rangecheckfailure
Mayaqua.lib(pem_lib.obj) : error LNK2001: Unresolvable external symbol __report_rangecheckfailure
Mayaqua.lib(cryptlib.obj) : error LNK2019: Unresolvable external symbol __stdio_common_vswprintf, which is in the function Referenced in OPENSSL_showfatal
Mayaqua.lib(cryptlib.obj) : error LNK2019: Unresolvable external symbol __stdio_common_vsscanf, which is in the function _ Referenced in vsscanf_l
Mayaqua.lib(v3_utl.obj) : error LNK2001: cannot The external symbol for parsing __stdio_common_vsscanf
D:\src_github\softethervpn\src\bin\vpntest_ x64.exe : fatal error LNK1120: 32 Unresolved External commands
LIBCMT.lib(sprintf.obj) : error LNK2005. sprintf has been defined in Mayaqua.lib(dso_win32.obj): sprintf has been defined in Mayaqua.lib(dso_win32.obj)
LIBCMT.lib(vsprintf.obj) : error LNK2005: _ vsprintf_l already in Mayaqua.lib(dso_win32.obj) define
LIBCMT.lib(printf.obj) : error LNK2005. printf has been defined in Mayaqua.lib(pqueue.obj): printf has been defined in Mayaqua.lib(pqueue.obj)


Translated with www.DeepL.com/Translator (free version)

**Comment by connLAN** [comment]
Hello @davidebeatrici ， my WIN7 OS native language is Chinese, so I use deepl.com translate it into english. I had tried to change WIN7 languge to English， but after reboot, it failed to run BuildAll.cmd.

**Comment by davidebeatrici** [comment]
You're building SoftEther VPN with the OpenSSL 1.1 headers and then linking to the 1.0.2 libraries.

You have to replace the headers as well (`src/Mayaqua/win32_inc/openssl`).

**Comment by connLAN** [comment]
I've tried many times as you told me above, but I am still failed. I noticed that  .lib files updated today, maybe you have some knowledge can share with us.

**Comment by davidebeatrici** [comment]
May I ask why you're attempting to link to OpenSSL 1.0.2?

**Comment by connLAN** [comment]
I want to replace openssl with gmssl(gmssl.org), so the first step I just replace it with my own. Compiled openssl libraries (libeay.lib and sslay.lib). As you know, as of openssl 1.1, the compiled libraries have been renamedince version 1.1.0 OpenSSL have changed their library names from. libeay32.dll -> libcrypto.dll
libeay32.dll -> libcrypto.dll
ssleay32.dll -> libssl.dll
And new engine libraries have been added, I don't know how they relate to each other. That's why I chose to use the 1.02 version.

Translated with www.DeepL.com/Translator (free version)
